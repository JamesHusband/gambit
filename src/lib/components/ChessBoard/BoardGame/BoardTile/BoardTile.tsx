import { cn } from "../../../../utils";
import { ChessPiece } from "../ChessPiece";
import type { Piece } from "../../../../types";

export const BoardTile = ({
  rowIndex,
  colIndex,
  piece,
}: {
  position: string;
  rowIndex: number;
  colIndex: number;
  piece?: Piece | null;
}) => {
  const isDark = (rowIndex + colIndex) % 2 === 1;

  return (
    <div
      data-testid="board-tile"
      className={cn(
        "flex items-center justify-center w-full h-full",
        isDark ? "bg-[#769656]" : "bg-[#eeeed2]"
      )}
    >
      {piece && <ChessPiece piece={piece} />}
    </div>
  );
};
