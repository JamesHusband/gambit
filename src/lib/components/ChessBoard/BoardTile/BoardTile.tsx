import { cn } from "../../../utils";

export const BoardTile = ({
  rowIndex,
  colIndex,
}: {
  position: string;
  rowIndex: number;
  colIndex: number;
}) => {
  const isDark = (rowIndex + colIndex) % 2 === 1;

  return (
    <div
      data-testid="board-tile"
      className={cn(
        "flex items-center justify-center w-full h-full",
        isDark ? "bg-[#769656]" : "bg-[#eeeed2]"
      )}
    />
  );
};
