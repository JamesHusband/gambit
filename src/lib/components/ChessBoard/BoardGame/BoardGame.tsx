import { BoardTile } from "./BoardTile";
import { useChessBoard } from "../../../hooks";

export const BoardGame = () => {
  const { board, getPieceAt } = useChessBoard();

  return (
    <div className="col-start-2 row-start-2 aspect-square w-full h-full">
      <div className="grid grid-cols-8 grid-rows-8 w-full h-full">
        {board.flat().map((square, index) => (
          <BoardTile
            key={square.position}
            position={square.position}
            rowIndex={Math.floor(index / 8)}
            colIndex={index % 8}
            piece={getPieceAt(square.position)}
          />
        ))}
      </div>
    </div>
  );
};
