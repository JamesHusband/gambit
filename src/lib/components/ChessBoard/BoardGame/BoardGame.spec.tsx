import { render } from "@testing-library/react";
import { BoardGame } from "./BoardGame";
import { BoardTile } from "./BoardTile";

jest.mock("./BoardTile", () => ({
  BoardTile: jest.fn(
    ({
      position,
      rowIndex,
      colIndex,
    }: {
      position: string;
      rowIndex: number;
      colIndex: number;
    }) => (
      <div data-testid="board-tile">
        {position} {rowIndex} {colIndex}
      </div>
    )
  ),
}));

jest.mock("../../../hooks", () => ({
  useChessBoard: () => ({
    board: [[{ position: "a8" }]],
  }),
}));

describe("BoardGame", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render", () => {
    const { container } = render(<BoardGame />);
    expect(container).toMatchSnapshot();
  });

  it("should render board tiles", () => {
    render(<BoardGame />);
    expect(BoardTile).toHaveBeenCalledTimes(1);

    const callArgs = (BoardTile as jest.Mock).mock.calls[0][0];
    expect(callArgs.position).toBe("a8");
    expect(callArgs.rowIndex).toBe(0);
    expect(callArgs.colIndex).toBe(0);
  });
});
