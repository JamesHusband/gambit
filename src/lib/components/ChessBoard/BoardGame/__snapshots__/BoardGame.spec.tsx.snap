// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BoardGame should render 1`] = `
<div>
  <div
    class="col-start-2 row-start-2 aspect-square w-full h-full"
  >
    <div
      class="grid grid-cols-8 grid-rows-8 w-full h-full"
    >
      <div
        data-testid="board-tile"
      >
        a8
         
        0
         
        0
      </div>
    </div>
  </div>
</div>
`;
